src/main/java/com/cachinginapp/enterprise/dao/ICacheDAO.java

```diff
@@ -8,4 +8,6 @@ public interface ICacheDAO {
     Cache save(Cache cache) throws Exception;
 
     List<Cache> fetchAll();
+
+    Cache fetchCacheById(int id);
```

Comment:
There is already a fetchCacheById in the service layer and it will need to call down to the DAO once you have a CacheService.

Microservices should have a single responsibility and by not separating the database call to the DAO that rule isn't being followed.
https://www.capitalone.com/tech/software-engineering/10-microservices-best-practices/